include/support.h

```diff
@@ -65,6 +65,27 @@ constexpr uint16_t dos_pagesize = 4096;
 // is in-range of a char and returns it as such.
 char int_to_char(int val);
 
+// A change in one of these 2 commits changed the behavior of toupper() on Windows with MSVC
```

Comment:
According to Microsoft, the `toupper` should only be called for ASCII characters, otherwise the results might be unexpected: https://learn.microsoft.com/en-us/cpp/c-runtime-library/reference/toupper-toupper-towupper-toupper-l-towupper-l?view=msvc-170#remarks

It seems when I started asking the standard C++ library about various locale formats, the Microsoft implementation (and only this one) changed the `toupper` behavior.

The comment needs to be updated, and we need to check all the other `toupper` calls in DOSBox Staging.

Reply:
Honestly, when weird shit like this surfaces I wish DOSBox was written in Java 😎 

No, really. Our build issues would go away and all the platform specific bullshit.

I'm like I'm transported back to the 60s when I see messy dodgy stuff like this... 😆 These are solved problems since Java 1.0 in the 90s.

Reply:
>  I wish DOSBox was written in Java

![stinky-clint-eastwood](https://github.com/user-attachments/assets/48bb0662-5180-41b9-a867-7bac50f51057)

Reply:
I looked at where `getch()` gets they key code from.  It's a PDCurses function that goes through a couple levels of parsing but ultimately the data comes from SDL events.  I don't think this should be affected by the locale changes so this is correct for the debugger.

I think it's best to merge this for now to fix the debugger problem.  I wrote that lengthy comment so we could go back and examine other uses of `toupper()` at a later date.

Reply:
I'll also echo what @FeralChild64 said in that this usecase of `toupper()` was probably a bug all along that just accidentally worked until the locale changed.  Not everything it passes is even a character (ex. the F5 key which is above the 255 that an 8 bit int can even hold).

If this turns out to be an edge case only applicable to the debugger, we can consider moving this function to `debug.cpp` and not worry about it elsewhere.